src/attached_probe.cpp

```diff
@@ -174,6 +184,11 @@ AttachedProbe::~AttachedProbe()
     case ProbeType::kretprobe:
       err = bpf_detach_kprobe(eventname().c_str());
       break;
+    case ProbeType::kfunc:
+    case ProbeType::kretfunc:
+      close(tracing_fd_);
+      err = bpf_detach_kfunc(progfd_, NULL);
```

Comment:
Looks like this is going to cause issues on systems that don't have this.

Reply:
ah right.. I'll detect these in cmake